src/com/facebook/buck/features/apple/project/ProjectGenerator.java

```diff
@@ -1721,8 +1721,11 @@ private PBXNativeTarget generateBinaryTarget(
           s -> extraSettingsBuilder.put("PRODUCT_MODULE_NAME", getModuleName(targetNode)));
 
       if (hasSwiftVersionArg && containsSwiftCode && isFocusedOnTarget) {
-        extraSettingsBuilder.put(
-            "SWIFT_OBJC_INTERFACE_HEADER_NAME", getSwiftObjCGeneratedHeaderName(buildTargetNode));
```

Comment:
Does the override really happen? If you put two entries with the same key, eventually an IllegalArgumentException will be thrown when build() is called.

Reply:
Sorry, seems like I wasn't clear enough. Look, I figured out, that Buck sets incorrect value for this parameter in xconfig files. I set this parameter in "configs" dictionary of "apple_binary" object to one I need. However, these two lines of code override my config during a build. So, I think this behaviour should be changed, because user's settings should have high priority over anything set by Buck itself.
Does it make sense?

Reply:
I meant extraSettingsBuilder is an ImmutableMap, and it does not allow set the same key twice. In the case here, setting  "SWIFT_OBJC_INTERFACE_HEADER_NAME" the second time is an error (extraSettingsBuilder.build() throws exception). So my understanding is overriding won't happen even before your change.